make vet does not work as it should in GNUmakefile

GNUmakefile has a directive called `vet` and currently it is not doing it's duty.

`make vet` checks if there are any suspicious constructs in the code, but at the moment we're only reporting error if it's exit code is equal to 1.


As per the Go team discussion this is not so: https://go-review.googlesource.com/c/go/+/101918 
> cmd/vet: only zero/non-zero exit status matters

> "go vet" exits with 2 on type-checking errors (because the compiler
> itself does so), but with 1 on vet errors.
>
> But we're really not trying to convey any nuance in vet's exit status.
> It's really just one bit: success or failure. Update the documentation
> so it doesn't claim anything subtler than this.

We currently have a few errors but `make vet` does not fail:
```
go vet .
# github.com/terraform-providers/terraform-provider-vcd/v2/vcd
vcd/resource_vcd_vapp_vm.go:153:14: result of fmt.Errorf call not used
vcd/resource_vcd_vapp_vm.go:240:13: result of fmt.Errorf call not used
```

I'm about to fix this.

--- a/GNUmakefile
+++ b/GNUmakefile
@@ -28,7 +28,7 @@ testacc: fmtcheck
 
 vet:
 	@echo "go vet ."
-	@go vet $$(go list ./... | grep -v vendor/) ; if [ $$? -eq 1 ]; then \
+	@go vet $$(go list ./... | grep -v vendor/) ; if [ $$? -ne 0 ]; then \
 		echo ""; \
 		echo "Vet found suspicious constructs. Please check the reported constructs"; \
 		echo "and fix them if necessary before submitting the code for review."; \

--- a/vcd/resource_vcd_vapp_vm.go
+++ b/vcd/resource_vcd_vapp_vm.go
@@ -150,7 +150,7 @@ func resourceVcdVAppVmCreate(d *schema.ResourceData, meta interface{}) error {
 			return err
 		}
 		if !isVappNetwork {
-			fmt.Errorf("vapp_network_name: %s is not found", vappNetworkName)
+			return fmt.Errorf("vapp_network_name: %s is not found", vappNetworkName)
 		}
 	}
 
@@ -237,7 +237,7 @@ func addVdcNetwork(d *schema.ResourceData, vdc govcd.Vdc, vapp govcd.VApp, vcdCl
 
 	net, err := vdc.FindVDCNetwork(networkNameToAdd)
 	if err != nil {
-		fmt.Errorf("network %s wasn't found as VDC network", networkNameToAdd)
+		return &types.OrgVDCNetwork{}, fmt.Errorf("network %s wasn't found as VDC network", networkNameToAdd)
 	}
 	vdcNetwork := net.OrgVDCNetwork
 

